### PR TITLE
Add UEFI parsing functionality

### PR DESCRIPTION
**Pull request by Ivanov1ch**
- [x] I have reviewed the [OFRAK contributor guide](https://ofrak.com/docs/contributor-guide/getting-started.html) and attest that this pull request is in accordance with it.

**One sentence summary of this PR (This should go in the CHANGELOG!)**
**WIP** Adds the ability to parse UEFI binaries. 
**Link to Related Issue(s)**
N/A
**Please describe the changes in your request.**
Adds the ability to unpack UEFI binaries using [UEFIExtract](https://github.com/LongSoft/UEFITool) and a corresponding test case built off UnpackAndVerify (see `test_uefi_component.py`). Also includes some small typo fixes. Current shortcomings/TODOs:
- [UEFI file format detection](https://wiki.osdev.org/UEFI) has not yet been implemented, so it is being detected as a PE and taking precendence
- Adds a dependency on UEFIExtract, which has a dependency that is currently missing from the Dockerstub (will not execute).\

**Anyone you think should look at this, specifically?**
@rbs-jacob

### CODE REVIEW

File: ofrak_core/Dockerstub [hunk 1]
```diff
@@ -63,3 +63,13 @@ WORKDIR /tmp
 RUN wget https://bootstrap.pypa.io/pip/get-pip.py && python3.9 get-pip.py && python3.7 get-pip.py && rm get-pip.py
 RUN python3.9 -m pip install jefferson
 WORKDIR /
+
+# Install UEFIExtract
+# TODO: This is broken - missing some deps to run
```
Comment (rbs-jacob) [inline ofrak_core/Dockerstub:68]: If you recompile so it uses the right version of GLIBC, then it should also work without the `$TARGETARCH` check.

File: ofrak_core/ofrak/core/uefi.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from ofrak.resource import Resource
+from ofrak.core.filesystem import File, Folder, FilesystemRoot, SpecialFileType
+
+
+from ofrak.model.component_model import ComponentExternalTool
+from ofrak.core.pe.model import Pe
+
+LOGGER = logging.getLogger(__name__)
+
+UEFIEXTRACT = ComponentExternalTool("uefiextract", "https://github.com/LongSoft/UEFITool", "--help")
```
Comment (rbs-jacob) [inline ofrak_core/ofrak/core/uefi.py:18]: Have you looked to see if this is on `apt` and `brew`? If so, would be good to add the respective package name(s) for those package managers here.

File: ofrak_core/test_ofrak/components/test_uefi_component.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        await root_resource.unpack_recursively()
+
+    async def get_descendants_to_verify(self, unpacked_root_resource: Resource) -> Dict:
+        result = {
+            await (
+                await descendent.view_as(FilesystemEntry)
+            ).get_path(): await descendent.get_data()
+            for descendent in await unpacked_root_resource.get_descendants()
+        }
+        print(result)
```
Comment (rbs-jacob) [inline ofrak_core/test_ofrak/components/test_uefi_component.py:67]:
```suggestion
```

File: ofrak_core/ofrak/core/uefi.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    async def unpack(self, resource: Resource, config=None):
+        ROM_FILE = "uefi.rom"
+
+        with tempfile.TemporaryDirectory() as temp_flush_dir:
+            os.chdir(temp_flush_dir)
+            await resource.flush_data_to_disk(ROM_FILE)
+            cmd = [
+                "uefiextract",
+                "uefi.rom",
```
Comment (rbs-jacob) [inline ofrak_core/ofrak/core/uefi.py:43]:
```suggestion
                ROM_FILE,
```

File: ofrak_core/ofrak/core/uefi.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    targets = (Uefi,)
+    children = (File, Folder, SpecialFileType)
+    external_dependencies = (UEFIEXTRACT,)
+
+    async def unpack(self, resource: Resource, config=None):
+        ROM_FILE = "uefi.rom"
+
+        with tempfile.TemporaryDirectory() as temp_flush_dir:
+            os.chdir(temp_flush_dir)
```
Comment (rbs-jacob) [inline ofrak_core/ofrak/core/uefi.py:39]: Add a comment noting why the `chdir` is necessary – don't want someone removing it in the future without knowing why it was put in. 

In general, code comments that explain what something does are not useful. But code comments that explain why something was done are extremely useful.

**Review by rbs-jacob — CHANGES_REQUESTED** [review]
In addition to the requested changes below, please also add a CHANGELOG entry for the addition of this functionality.

File: ofrak_core/ofrak/core/uefi.py (hunk 2 above)
Comment (Ivanov1ch) [reply ofrak_core/ofrak/core/uefi.py:18]: Not on either

**Review by rbs-jacob — APPROVED** [review]

**Pull request merged**